Resolved issue #6. Added configuration for service hostname and port.

mpeng3 opened the pull request:
Resolves issue [#6].

I created a new "httpConnection" configuration class and corresponding section in the .yml file.  Some of these properties may be redundant (esp. port #, but auto-fetching them [appears to be slightly complicated](https://github.com/dropwizard/dropwizard/issues/745)).

I would appreciate feedback regarding whether it is better to use pre-existing configuration items, or create new ones.  Using pre-existing config entries means less potential duplication of config information.  However, as I believe you were trying to point out, in cases where the service is sitting inside of a complex environment, pre-existing config parameters may not behave how we want.

mpeng3 committed:
Resolved issue #6. Added configuration for service hostname and port.

StdVectorBool commented on UrlShortener/shrtn.yml:
```diff
@@ -1,3 +1,10 @@
+  host: localhost
+
+  # the port to use
+  port: 8080
```
As you noted, there is an official [property](http://www.dropwizard.io/0.9.3/docs/manual/configuration.html#man-configuration-http) for the application port, so duplicating it adds room for error.  Weight "slightly complicated code that is automatically correct" vs "manual configuration that is silently error prone".

Removing 'port' would make httpConnection.host a one-item hierarchy so perhaps just "hostname" at the top of Configuration would be better if you go down that route.

In a complex environment, the app probably sits behind a load balancer that uses standard HTTP (80 or HTTPS 443) to be friendly to the user (the browser just needs "http://" or "https://" to imply the port).

StdVectorBool committed:
Test configurable host names and port in shortener responses. Add
examples of mocking and fluent assertions.

mpeng3 committed:
Resolved Issue #6. Hostname property moved to top level.

Duplicate port property has been removed.  Added a convenience function
to set the port programmatically and fetch the port at runtime.
